test/functional/feature_csv_activation.py

```diff
@@ -175,15 +183,15 @@ def run_test(self):
         self.log.info("Generate blocks in the past for coinbase outputs.")
         long_past_time = int(time.time()) - 600 * 1000  # enough to build up to 1000 blocks 10 minutes apart without worrying about getting into the future
         self.nodes[0].setmocktime(long_past_time - 100)  # enough so that the generated blocks will still all be before long_past_time
-        self.coinbase_blocks = self.nodes[0].generate(1 + 16 + 2 * 32 + 1)  # 82 blocks generated for inputs
+        self.coinbase_blocks = self.nodes[0].generate(1 + 16 + 2 * 32 + 2)  # 83 blocks generated for inputs
         self.nodes[0].setmocktime(0)  # set time back to present so yielded blocks aren't in the future as we advance last_block_time
-        self.tipheight = 82  # height of the next block to build
+        self.tipheight = 83  # height of the next block to build
         self.last_block_time = long_past_time
         self.tip = int(self.nodes[0].getbestblockhash(), 16)
         self.nodeaddress = self.nodes[0].getnewaddress()
 
         # Activation height is hardcoded
-        test_blocks = self.generate_blocks(345)
+        test_blocks = self.generate_blocks(344)
```

Comment:
what's the justification for this change? Is the CSV activation at block 344 instead of 345?

Reply:
The CSV activation height didn't change (is hardcoded as 432 in regtest mode), but since we are generating one coinbase block more, one block less has to be generated at this point to reach the desired height -- which turned out to be the CSV activation height minus five. Tried to get rid of most magic numbers in commit https://github.com/bitcoin/bitcoin/pull/17921/commits/93f5ee06e5cc620fdea5e990b46f3d23165cd9b0.

Reply:
Ah! Then maybe `CSV_ACTIVATION_HEIGHT - TX_BLOCK_COUNT` would be better?